### DISCUSSION TITLE
customManagers: Detecting currentValue and depName in different files possible?

### DISCUSSION THREAD

**Discussion by kastl-ars** [renovatebot/renovate#29294] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Forgejo with 37.377.4

### Please tell us more about your question or problem

* I have multiple charts in subfolders of a chart directory.
* I have a working `customManager` to change the `appVersion` in the `Chart.yaml` file, but currently only by hardcoding the `depNameTemplate` (and by running a postUpgradeTask, but that does not matter here)
* I would like to let renovate detect that, so I do not have to add a customManager for each single chart
* The docker image reference aka `depName` is in the `values.yaml` only, while there is no tag in this file, as the `appVersion` from `Chart.yaml` is being used as the image tag

I came up with something like this, but this does not do anything unfortunately:

```
  "customManagers": [
    {
      "customType": "regex",
      "fileMatch": [
        "^charts\\/my-chart\\/Chart\\.yaml$",
        "^charts\\/my-chart\\/values\\.yaml$"
      ],
      "matchStrings": [
        "^appVersion:\\s+(?<currentValue>.*)",
        "\\s+repository:\\s+(?<depName>.*)"
      ],
      "matchStringsStrategy": "combination",
      "datasourceTemplate": "docker"
    }
  ],
```

I tried different values of `matchStringsStrategy`, but without success. From the docs and other issues it seems to me that this does not work as only one of the matchStrings is found for each file.

The automatic helm-values manager does not work, as this expects an image tag in the `values.yaml`, as far as I am aware...

Maybe someone has a working example, I cannot imagine that I am the only one having this itch to scratch...

Kind Regards
Johannes

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by viceice** [comment, marked as answer]
it's not possible, add a yaml comment to the chart with the depName and match it in same file

**Reply by kastl-ars** [reply]
I found the comment and tried it, but was hoping it **was** possible nevertheless.

Thanks for the quick reply, Michael!
